app/src/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -241,21 +246,21 @@
      */
     @Override
     @SuppressLint("NewApi")
-    public void onCreate(Bundle savedInstanceState) {
+    public void onCreate(final Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);        
 
         // See if this form needs GPS to be turned on
         mNoGPSReceiver = new BroadcastReceiver() {
             @Override
-            public void onReceive(Context context, Intent intent) {
+            public void onReceive(final Context context, final Intent intent) {
                 context.removeStickyBroadcast(intent);
-                LocationManager manager = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
-                Set<String> providers = GeoUtils.evaluateProviders(manager);
+                final LocationManager manager = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
```

Comment:
These 'final' changes are making the diff hard to read. Are they stylistic or functional?